lib/resources/users.js

```diff
@@ -101,36 +130,27 @@ module.exports = (service, endpoint) => {
     Users.getByActorId(params.id)
       .then(getOrNotFound)
       .then((user) => auth.canOrReject('user.update', user.actor)
-        .then(() => User.fromApi(body))
+        .then(() => {
+          if (oidc.isEnabled()) {
+            // Don't allow modifying email or password for users when using OIDC.
+            // This can't be done in /frames/ as it does not allow conditional filtering.
+            const { email, password, ...filtered } = body;
+            return filtered;
```

Comment:
The release criteria suggest that admins still need to be able to update users' email addresses. @sadiqkhoja left a comment about that in the release criteria, but if we do decide to run with that behavior, this logic would need to change.

Also, `password` isn't a `writable` field on the frame, so I don't think it needs to be filtered out here: `fromApi()` should always filter it out. This endpoint can't be used to change a password: that's a different endpoint.

Reply:
So can this whole block be deleted?

Reply:
I think there does need to be something here. If SSO is enabled, we don't want non-admin users to use this endpoint to update their own email. Note that in `auth`, users can `user.update` themselves even if they can't `user.update` other users:

https://github.com/getodk/central-backend/blob/cedadf298742dcf4c3c320ac27f2d0fff95b7476/lib/model/query/auth.js#L26-L29

Non-admin users should still be able to use this endpoint to update their own display name.

I think this means that we should filter out `email` if users can't `user.update` users in general. I think you can check for that by calling `auth.can('user.update', User.species)`, which returns a promise.

Reply:
I personally have no preference. @issa-tseng maybe you have strong feelings?

I think we could say admins can change their own email because they should have a lot more context on the implications. This is also more helpful in contexts in which there’s only one site wide admin. 

It would also be very defensible to say admins need to use the command line or talk to another admin to get their email address changed. 

Whatever it is, it should be in the docs.

Reply:
Whatever the behavior is, we should also make sure that Frontend matches it. Right now, Frontend will allow an admin to change their own email.